tests/general/FileModeBearTest.py

```diff
@@ -0,0 +1,212 @@
+import os
+import platform
+import stat
+
+from queue import Queue
+
+from bears.general.FileModeBear import FileModeBear
+from coalib.testing.LocalBearTestHelper import LocalBearTestHelper
+from coalib.results.Result import RESULT_SEVERITY, Result
+from coalib.settings.Section import Section
+from coalib.settings.Setting import Setting
+
+
+def get_testfile_path(file):
+    return os.path.join(os.path.dirname(__file__),
+                        'filemode_test_files', file)
+
+
+FILE_PATH = get_testfile_path('test_file.txt')
+
+
+class FileModeBearTest(LocalBearTestHelper):
+
+    def setUp(self):
+        self.section = Section('')
+        self.uut = FileModeBear(self.section, Queue())
+
+    def test_r_to_r_permissions(self):
+        os.chmod(FILE_PATH, stat.S_IRUSR)
+        self.section.append(Setting('filemode', 'r'))
+        self.check_results(
+            self.uut,
+            [],
+            [],
+            filename=FILE_PATH,
+            )
+
+    def test_w_to_w_permissions(self):
+        os.chmod(FILE_PATH, stat.S_IWUSR)
+        self.section.append(Setting('filemode', 'w'))
+        self.check_results(
+            self.uut,
+            [],
+            [],
+            filename=FILE_PATH,
+            )
+
+    def test_x_to_x_permissions(self):
+        os.chmod(FILE_PATH, stat.S_IXUSR)
+        if platform.system() != 'Windows':
```

Comment:
Mention the reason wherever you have done this.

Reply:
No.  Get rid of it.  If this bear doesnt support Windows, that fact shouldnt be hidden inside the logic of the tests.

A `.coafile` is a cross-platform specification, and it must not cause failures on only one platform.

Reply:
Perhaps create a separate test class for executable permissions, and the class should be skipped on Windows for tests which dont work on Windows, and potentially we need a second extra test class which does Windows specific testing to show that specifying executable bits in the .coafile results in a mild note in the coala log, but does not cause it to fail.